src/active-response/wazuh-slack.c

```diff
@@ -0,0 +1,265 @@
+                                        );
+
+        memset(str_rule, '\0', LOGSIZE);
+        strcpy(str_rule, temp_line);
+    }
+
+    memset(color, '\0', 100);
+    if (level <= 4) {
+        strcpy(color, "good");
+        cJSON_AddStringToObject(item_objects, "color", "good");
+    } else if (level >= 5 && level <= 7){
+        strcpy(color, "warning");
+        cJSON_AddStringToObject(item_objects, "color", "warning");
+    } else {
+        strcpy(color, "danger");
+        cJSON_AddStringToObject(item_objects, "color", "danger");
+    }
+
+
+    cJSON_AddStringToObject(item_objects, "pretext", "WAZUH Alert");
+    cJSON_AddStringToObject(item_objects, "title", rule_description_json != NULL ? rule_description_json->valuestring : "N/A");
+
+    // Detect full log
+    full_log_json = cJSON_GetObjectItem(alert, "full_log");
+    cJSON_AddStringToObject(item_objects, "text", full_log_json != NULL ? full_log_json->valuestring : "");
+
+    // Detect agent
+    agent_json = cJSON_GetObjectItem(alert, "agent");
+    if (agent_json && (agent_json->type == cJSON_Object)) {
+        cJSON *agent_id_json = NULL;
+        cJSON *agent_name_json = NULL;
+
+        item_agent = cJSON_CreateObject();
+
+        // Detect Agent ID
+        agent_id_json = cJSON_GetObjectItem(agent_json, "id");
+
+        // Detect Agent name
+        agent_name_json = cJSON_GetObjectItem(agent_json, "name");
+
+        memset(temp_line, '\0', LOGSIZE);
+        snprintf(temp_line, LOGSIZE -1, "(%s) - %s",
+                                        agent_id_json != NULL ? agent_id_json->valuestring : "N/A",
+                                        agent_name_json != NULL ? agent_name_json->valuestring : "N/A"
+                                        );
+
+        cJSON_AddStringToObject(item_agent, "title", "Agent");
+        cJSON_AddStringToObject(item_agent, "value", temp_line);
+        cJSON_AddItemToArray(fields_list, item_agent);
+    }
+
+    // Detect agentless
+    agentless_json = cJSON_GetObjectItem(alert, "agentless");
+    if (agentless_json && (agentless_json->type == cJSON_Object)) {
+        cJSON *agentless_host_json = NULL;
+        item_agentless = cJSON_CreateObject();
+
+        // Detect Agentless host
+        agentless_host_json = cJSON_GetObjectItem(agentless_json, "host");
+
+        cJSON_AddStringToObject(item_agentless, "title", "Agentless Host");
+        cJSON_AddStringToObject(item_agentless, "value", agentless_host_json != NULL ? agentless_host_json->valuestring : "N/A");
+        cJSON_AddItemToArray(fields_list, item_agentless);
+    }
+
+    location_json = cJSON_GetObjectItem(alert, "location");
+    item_location = cJSON_CreateObject();
+    cJSON_AddStringToObject(item_location, "title", "Location");
+    cJSON_AddStringToObject(item_location, "value", location_json != NULL ? location_json->valuestring : "N/A");
+    cJSON_AddItemToArray(fields_list, item_location);
+
+    item_rule = cJSON_CreateObject();
```

Comment:
Move these 5 lines inside [this if](https://github.com/wazuh/wazuh/pull/7297/files#diff-77fc4cbe7be48b8998c4e3b1041ed977fbed191645c92954dd4719fafd09a2b2R157) so you can delete `str_rule` variable.